lib/adapters/http.js

```diff
@@ -75,12 +120,14 @@ function setProxy(options, configProxy, location) {
       if (proxy.auth.username || proxy.auth.password) {
         proxy.auth = (proxy.auth.username || '') + ':' + (proxy.auth.password || '');
       }
-      const base64 = Buffer
-        .from(proxy.auth, 'utf8')
-        .toString('base64');
-      options.headers['Proxy-Authorization'] = 'Basic ' + base64;
     }
 
+    options._httpsOverHttp = proxy.protocol === "http:" && options.protocol === "https:";
```

Comment:
What about https over https? You still need a tunnel in that case because you need end-to-end encryption with the intended recipient.

I'm wondering if I might not open a PR with our own solution which relied un the underlying agents (which I mentioned [here](https://github.com/axios/axios/issues/4318#issuecomment-1176692529)). 

You could also check out my fork directly : https://github.com/mbargiel/axios/compare/e6f9026d51d5e1386bc6935e563d95c375c45d02...feature/fix-proxy/http-connect

Reply:
The nice thing with agents is that you don't have to worry about problems with follow-redirects: the tunnel is established in a lower abstraction layer, and once Axios has the socket, it's established properly to the end server. Redirects work over the tunnel.